app/models/customer_account.rb

```diff
@@ -65,6 +65,14 @@ def self.find_by_reference(reference)
       nil
     end
 
+    # Find customer account by password reset token provided in email's link
+    # Used in reset password scenario
+    def self.find_by_token(token)
```

Comment:
I have commented on this in the platform - just wondering if we are creating an inconsistency by doing this as opposed to expecting the developers to use find("token:#{token}") ?